src/lib/y2network/dialogs/wireless_wep_keys.rb

```diff
@@ -0,0 +1,253 @@
+require "cwm/common_widgets"
+
+module Y2Network
+  module Dialogs
+    # Dialog to manage WEP keys
+    class WirelessWepKeys < CWM::Dialog
+      # @param builder [Y2network::InterfaceConfigBuilder]
+      def initialize(builder)
+        textdomain "network"
+        @builder = builder
+      end
+
+      def title
+        _("Wireless Keys")
+      end
+
+      def help
+        _(
+          "<p>In this dialog, define your WEP keys used\n" \
+            "to encrypt your data before it is transmitted. You can have up to four keys,\n" \
+            "although only one key is used to encrypt the data. This is the default key.\n" \
+            "The other keys can be used to decrypt data. Usually you have only\n" \
+            "one key.</p>"
+        )
+      end
+
+      def contents
+        HBox(
+          HSpacing(5),
+          VBox(
+            VSpacing(1),
+            # Frame label
+            Frame(
+              _("WEP Keys"),
+              HBox(
+                HSpacing(3),
+                VBox(
+                  VSpacing(1),
+                  # ComboBox label
+                  Left(WEPKeyLength.new(@builder)),
+                  VSpacing(1),
+                  WEPKeys.new(@builder),
+                  VSpacing(1)
+                ),
+                HSpacing(3)
+              )
+            ),
+            VSpacing(1)
+          ),
+          HSpacing(5)
+        )
+      end
+
+      # Always open new dialog to work properly in sequence
+      def should_open_dialog?
+        true
+      end
+
+      class WEPKeyLength < CWM::ComboBox
+        def initialize(builder)
+          textdomain "network"
+
+          @builder = builder
+        end
+
+        def items
+          [
+            ["64", "64"],
+            ["128", "128"]
+          ]
+        end
+
+        def init
+          length_s = @builder.key_length.to_s
+          self.value = length_s.empty? ? "128" : length_s
+        end
+
+        def store
+          @builder.key_length = value.to_i
+        end
+
+        def label
+          _("&Key Length")
+        end
+
+        def help
+          _(
+            "<p><b>Key Length</b> defines the bit length of your WEP keys.\n" \
+              "Possible are 64 and 128 bit, sometimes also referred to as 40 and 104 bit.\n" \
+              "Some older hardware might not be able to handle 128 bit keys, so if your\n" \
+              "wireless LAN connection does not establish, you may need to set this\n" \
+              "value to 64.</p>"
+          )
+        end
+      end
+
+      class WEPKeys < CWM::CustomWidget
+        def initialize(settings)
+          textdomain "network"
+          @settings = settings
+        end
+
+        def contents
+          VBox(
+            Table(
+              Id(:wep_keys_table),
+              Header(
+                # Table header label
+                # Abbreviation of Number
+                _("No."),
+                # Table header label
```

Comment:
Is it needed to put "# Table header label" for each header? :)

Reply:
it is hint for translators. I worry it won't work if it is there just once before first one.

Reply:
Oh, I missed the "# TRANSLATORS:" prefix.